spec/services/process_submission_service_spec.rb

```diff
@@ -479,155 +479,50 @@
+              'text/plain' => 'https://tools.ietf.org/rfc/rfc2324.txt'
             },
-            {
-              'type' => 'filestore',
-              'mimetype' => 'image/png',
-              'url' => 'http://fb-user-filestore-api-svc-test-dev.formbuilder-platform-test-dev//service/ioj/user/a239313d-4d2d-4a16-b5ef-69d6e8e53e86/28d-dae59621acecd4b1596dd0e96968c6cec3fae7927613a12c357e7a62e11877d8',
-              'filename' => 'image2.png'
-            }
-          ]
-        }
-      end
-
-      before do
-        allow(Submission).to receive(:find).and_return(submission)
-        allow(submission).to receive(:id).and_return('id-of-submission')
-      end
-
-      it 'converts output urls to absolute urls' do
-        url = submission.detail_objects.dig(0).attachments.dig(0, 'url')
-        expect(url).to start_with('http')
-      end
-
-      it 'attaches filestore attachments' do
-        expect(DownloadService).to receive(:download_in_parallel)
-          .with(headers: {
-                  'x-encrypted-user-id-and-token' => 'encrypted_user_id_and_token'
-                }, urls: [
-                  'http://fb-user-filestore-api-svc-test-dev.formbuilder-platform-test-dev//service/ioj/user/a239313d-4d2d-4a16-b5ef-69d6e8e53e86/28d-dae59621acecd4b1596dd0e96968c6cec3fae7927613a12c357e7a62e11877d8',
-                  'http://service-slug.formbuilder-services-test:3000/api/submitter/pdf/default/guid1.pdf'
-                ]).and_return(
-                  'http://fb-user-filestore-api-svc-test-dev.formbuilder-platform-test-dev//service/ioj/user/a239313d-4d2d-4a16-b5ef-69d6e8e53e86/28d-dae59621acecd4b1596dd0e96968c6cec3fae7927613a12c357e7a62e11877d8' => '/tmp/filestore.png',
-                  'http://service-slug.formbuilder-services-test:3000/api/submitter/pdf/default/guid1.pdf' => '/tmp/output.pdf'
-                )
-
-        # only testing file attachments here
-        allow(subject).to receive(:retrieve_mail_body_parts).and_return([])
-
-        expected_attachments = [
-          Attachment.new(
-            path: '/tmp/output.pdf',
-            type: 'output',
-            mimetype: 'application/pdf',
-            url: 'http://service-slug.formbuilder-services-test:3000/api/submitter/pdf/default/guid1.pdf',
-            filename: 'form1'
-          ),
-          Attachment.new(
-            path: '/tmp/filestore.png',
-            type: 'filestore',
-            mimetype: 'image/png',
-            url: 'http://fb-user-filestore-api-svc-test-dev.formbuilder-platform-test-dev//service/ioj/user/a239313d-4d2d-4a16-b5ef-69d6e8e53e86/28d-dae59621acecd4b1596dd0e96968c6cec3fae7927613a12c357e7a62e11877d8',
-            filename: 'image2.png'
-          )
-        ]
-
-        subject.send(:attachments, submission.detail_objects[0]).each_with_index do |a, i|
-          %i[type filename url mimetype path].each do |k|
-            expect(a.send(k)).to eql(expected_attachments[i].send(k))
-          end
-        end
-
```

Comment:
these tests are redundant to what is tested above. they are also problematic in that this assertion is causing side effects in the code.